### PR TITLE
Caption fixups for Lower Your Guards

### PR DESCRIPTION
**Pull request by RyanGlScott**
cc @sgraf812

### CODE REVIEW

File: session4/session4.ass [hunk 1]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
-Dialogue: 0,1:49:18.36,1:49:22.12,Default,,0,0,0,,we abandoned all structural\N pattern matching like if f here
+Dialogue: 0,1:49:18.36,1:49:22.12,Default,,0,0,0,,we abandoned all structural\N pattern matching like in f here
 Dialogue: 0,1:49:22.12,1:49:25.56,Default,,0,0,0,,and desugar function\N definitions into guard trees.
 Dialogue: 0,1:49:27.08,1:49:29.40,Default,,0,0,0,,Then we do coverage checking\N on these guard trees
 Dialogue: 0,1:49:29.40,1:49:34.08,Default,,0,0,0,,which is split in two\N simple functions A and U.
 Dialogue: 0,1:49:35.56,1:49:39.32,Default,,0,0,0,,A returns an annotated tree\N decorated with refinement types
-Dialogue: 0,1:49:39.32,1:49:42.32,Default,,0,0,0,,that capture redundancies\N and inaccessibility information.
-Dialogue: 0,1:49:43.32,1:49:45.24,Default,,0,0,0,,If you want to know about these part
+Dialogue: 0,1:49:39.32,1:49:42.32,Default,,0,0,0,,that capture redundancy\N and inaccessibility information.
+Dialogue: 0,1:49:43.32,1:49:45.24,Default,,0,0,0,,If you want to know about this part,
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:49:43.32,1:49:45.24,Default,,0,0,0,,If you want to know more about this part,
```

File: session4/session4.ass [hunk 2]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
 Dialogue: 0,1:50:07.84,1:50:09.84,Default,,0,0,0,,Green indicates source syntax.
 Dialogue: 0,1:50:09.84,1:50:13.04,Default,,0,0,0,,You can see that fancy makes\N use of constructor patterns
-Dialogue: 0,1:50:13.04,1:50:14.68,Default,,0,0,0,,and even to language extensions
-Dialogue: 0,1:50:15.40,1:50:17.68,Default,,0,0,0,,a bang pattern that\N forces the match thing
-Dialogue: 0,1:50:18.44,1:50:21.96,Default,,0,0,0,,and a view pattern that\N applies the match thing to g
-Dialogue: 0,1:50:23.44,1:50:25.64,Default,,0,0,0,,and matches\N the results against True.
-Dialogue: 0,1:50:29.12,1:50:32.04,Default,,0,0,0,,In yellow you see fancy\N desugaring to guard trees.
+Dialogue: 0,1:50:13.04,1:50:14.68,Default,,0,0,0,,and even two language extensions:
+Dialogue: 0,1:50:15.40,1:50:17.68,Default,,0,0,0,,a bang pattern that\N forces the matched thing,
+Dialogue: 0,1:50:18.44,1:50:21.96,Default,,0,0,0,,and a view pattern that\N applies the matched thing to g
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:50:18.44,1:50:21.96,Default,,0,0,0,,and a view pattern that\N applies g to the matched thing
```

I know this isn't entirely faithful to the audio, but as Simon suggested this is what I should have said.

File: session4/session4.ass [hunk 3]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
+Dialogue: 0,1:50:23.44,1:50:25.64,Default,,0,0,0,,and matches\N the result against True.
+Dialogue: 0,1:50:29.12,1:50:32.04,Default,,0,0,0,,In yellow you see fancy's\N desugaring to guard trees.
 Dialogue: 0,1:50:33.28,1:50:35.24,Default,,0,0,0,,Here is the syntax\N definition for reference.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:50:33.28,1:50:35.24,Default,,0,0,0,,Here is their syntax\N definition for reference.
```

File: session4/session4.ass [hunk 4]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
+Dialogue: 0,1:51:42.48,1:51:45.88,Default,,0,0,0,,like the one binding xs to\N the temporary t2 here.
+Dialogue: 0,1:51:49.88,1:51:53.28,Default,,0,0,0,,We found similar desugarings for\N all the pattern-match syntax
 Dialogue: 0,1:51:53.28,1:51:54.28,Default,,0,0,0,,of modern Haskell.
 Dialogue: 0,1:51:55.00,1:51:58.04,Default,,0,0,0,,Also note that desugaring is\N for coverage checking only.
 Dialogue: 0,1:51:58.04,1:51:59.44,Default,,0,0,0,,There is no code\N generation involved.
-Dialogue: 0,1:52:03.20,1:52:08.08,Default,,0,0,0,,Alright, next up characterizing\N the set of uncovered values
+Dialogue: 0,1:52:03.20,1:52:08.08,Default,,0,0,0,,Alright, next up: characterizing\N the set of uncovered values
 Dialogue: 0,1:52:08.08,1:52:10.44,Default,,0,0,0,,that we can report\N uncovered patterns for.
 Dialogue: 0,1:52:11.68,1:52:16.16,Default,,0,0,0,,We compute the set by gradually\N refining the set of possible values
 Dialogue: 0,1:52:17.16,1:52:21.04,Default,,0,0,0,,as a fraction of them fall through\N from one clause to the next.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:52:17.16,1:52:21.04,Default,,0,0,0,,as a fraction of them falls through\N from one clause to the next.

```

File: session4/session4.ass [hunk 5]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
 Dialogue: 0,1:52:23.04,1:52:27.36,Default,,0,0,0,,How do we represent such a set with\N possibly infinitely many values?
 Dialogue: 0,1:52:28.84,1:52:30.52,Default,,0,0,0,,We pick refinement\N types for the job.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:52:28.84,1:52:30.52,Default,,0,0,0,,We picked refinement\N types for the job.

```

File: session4/session4.ass [hunk 6]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
-Dialogue: 0,1:53:55.04,1:53:57.28,Default,,0,0,0,,you compute the subset of theta
+Dialogue: 0,1:53:55.04,1:53:57.28,Default,,0,0,0,,U computes the subset of Θ
 Dialogue: 0,1:53:57.28,1:54:00.32,Default,,0,0,0,,whose values do not\N match the guard tree t.
-Dialogue: 0,1:54:02.32,1:54:05.60,Default,,0,0,0,,When we reach the right hand side,\N every incoming value is covered.
+Dialogue: 0,1:54:02.32,1:54:05.60,Default,,0,0,0,,When we reach the right-hand side,\N every incoming value is covered.
 Dialogue: 0,1:54:05.60,1:54:07.96,Default,,0,0,0,,So we return\N the empty refinement type.
-Dialogue: 0,1:54:08.96,1:54:11.16,Default,,0,0,0,,Branches model fall\N through semantics.
+Dialogue: 0,1:54:08.96,1:54:11.16,Default,,0,0,0,,Branches model fall-\N through semantics.
 Dialogue: 0,1:54:11.16,1:54:13.48,Default,,0,0,0,,Only values that fall\N through the first branch
 Dialogue: 0,1:54:14.48,1:54:16.40,Default,,0,0,0,,and fall through the second branch.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:54:14.48,1:54:16.40,Default,,0,0,0,,can fall through the second branch.

```

File: session4/session4.ass [hunk 7]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
-Dialogue: 0,1:54:27.32,1:54:30.68,Default,,0,0,0,,The pattern guard matching\N against constructor k
-Dialogue: 0,1:54:30.68,1:54:33.32,Default,,0,0,0,,will fall through when x is not k.
-Dialogue: 0,1:54:35.56,1:54:39.44,Default,,0,0,0,,Also everything not covered\N by the subtree is uncovered
+Dialogue: 0,1:54:27.32,1:54:30.68,Default,,0,0,0,,A pattern guard matching\N against constructor K
+Dialogue: 0,1:54:30.68,1:54:33.32,Default,,0,0,0,,will fall through when x is not K.
+Dialogue: 0,1:54:35.56,1:54:39.44,Default,,0,0,0,,Also everything not covered\N by the subtree is uncovered,
 Dialogue: 0,1:54:39.44,1:54:43.08,Default,,0,0,0,,so we adjoin the two refinement types.
-Dialogue: 0,1:54:44.08,1:54:45.84,Default,,0,0,0,,And that is all there is\N to coverage checking.
-Dialogue: 0,1:54:47.32,1:54:51.80,Default,,0,0,0,,The next step is to generate\N inhabitants of the refinement types
+Dialogue: 0,1:54:44.08,1:54:45.84,Default,,0,0,0,,And that'ss all there is\N to coverage checking.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:54:44.08,1:54:45.84,Default,,0,0,0,,And that's all there is\N to coverage checking.

```

File: session4/session4.ass [hunk 8]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
+Dialogue: 0,1:55:16.68,1:55:19.68,Default,,0,0,0,,or an SMT solver directly,\N to generate inhabitants.
 Dialogue: 0,1:55:19.68,1:55:22.52,Default,,0,0,0,,But we didn't do that, because
-Dialogue: 0,1:55:22.52,1:55:27.20,Default,,0,0,0,,reflecting GHC's type\N constraint solver into SMT logic,
-Dialogue: 0,1:55:27.20,1:55:29.08,Default,,0,0,0,,it's quite an undertaking.
-Dialogue: 0,1:55:29.08,1:55:33.52,Default,,0,0,0,,And we think that FFI calls\N and serialization affects
-Dialogue: 0,1:55:33.52,1:55:35.84,Default,,0,0,0,,induce too much overhead.
-Dialogue: 0,1:55:35.84,1:55:37.44,Default,,0,0,0,,So, we bit the bullet,
+Dialogue: 0,1:55:22.52,1:55:27.20,Default,,0,0,0,,reflecting GHC's type-\N constraint solver into SMT logic
+Dialogue: 0,1:55:27.20,1:55:29.08,Default,,0,0,0,,is quite an undertaking.
+Dialogue: 0,1:55:29.08,1:55:33.52,Default,,0,0,0,,And we think that FFI calls\N and serialization effects
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:55:29.08,1:55:33.52,Default,,0,0,0,,And we think that FFI calls\N and serialization of facts

```

File: session4/session4.ass [hunk 9]
```diff
@@ -1910,241 +1910,241 @@ Dialogue: 0,1:43:07.92,1:43:11.88,Default,,0,0,0,,It is based on work by Sebasti
...
+Dialogue: 0,1:55:33.52,1:55:35.84,Default,,0,0,0,,incur too much overhead.
+Dialogue: 0,1:55:35.84,1:55:37.44,Default,,0,0,0,,So, we bit the bullet
 Dialogue: 0,1:55:37.44,1:55:40.76,Default,,0,0,0,,and just wrote our own ad\N hoc generator function.
```
Comment (sgraf812) [inline session4/session4.ass]:
```suggestion
Dialogue: 0,1:55:37.44,1:55:40.76,Default,,0,0,0,,and just wrote our own \N ad-hoc generator function.

```

**Review by sgraf812 — CHANGES_REQUESTED** [review]
Thanks for doing this! I proof-read along with the audio and added my own changes as suggestions. Could you please accept them, too?

**Comment by RyanGlScott** [comment]
Thanks, @sgraf812!

I think this should be ready to go now, but please shout if I messed up any of the `.ass`/`.srt` conventions.

**Comment by sgraf812** [comment]
LGTM!

**Comment by lkuper** [comment]
Thanks!  (There was no need to update the .srt too, but it's not a problem that you did.)

**Pull request merged**